syft/pkg/apk_metadata.go

```diff
@@ -121,23 +124,44 @@ func (m ApkMetadata) OwnedFiles() (result []string) {
 	return result
 }
 
-func (m ApkMetadata) Upstream() string {
+type UpstreamCandidate struct {
+	Name string
+	Type Type
+}
+
+func (m ApkMetadata) UpstreamCandidates() (candidates []UpstreamCandidate) {
```

Comment:
Yeah, I would have liked that; however, it's also needed for the purl.

Reply:
Would still have to be exported though, right?

Reply:
I missed the usage in purl processing. I take back my comment then. I think in the future this will be in a better position to refactor: later we'll be moving the CPE generation logic back to where packages are created (near the catalogers) which means that we could remove this from the metadata object at that time.